packages/expo-clipboard/ios/ClipboardPasteButton.swift

```diff
@@ -0,0 +1,178 @@
+  override func paste(itemProviders: [NSItemProvider]) {
+    if #available(iOS 14.0, *) {
+      for provider in itemProviders {
+        if provider.hasItemConformingToTypeIdentifier(UTType.image.identifier) {
+          _ = provider.loadObject(ofClass: UIImage.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted image")
+              return
+            }
+            self.processImage(item: data)
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.url.identifier) {
+          _ = provider.loadObject(ofClass: URL.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+
+            self.onPastePressed([
+              "type": "text",
+              "text": data?.absoluteString
+            ])
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.html.identifier) && acceptedContentTypes.contains(.html) {
+          _ = provider.loadObject(ofClass: String.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+            self.processHtml(data: data)
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.utf8PlainText.identifier) && !acceptedContentTypes.contains(.html) {
+          _ = provider.loadObject(ofClass: String.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+
+            guard let data = data as? String else {
+              log.error("Failed to read text data")
+              return
+            }
+
+            self.onPastePressed([
+              "type": "text",
+              "text": data
+            ])
+          }
+        }
+      }
+    }
+  }
+
+  @available(iOS 14.0, *)
+  private func setContentTypes() {
+    if acceptedContentTypes.isEmpty {
+      pasteConfiguration = UIPasteConfiguration(acceptableTypeIdentifiers: [
+        UTType.utf8PlainText.identifier,
+        UTType.image.identifier
+      ])
+    } else {
+      pasteConfiguration = UIPasteConfiguration(acceptableTypeIdentifiers: acceptedContentTypes.map {
+        $0.typeIdentifier()
+      })
+    }
+  }
+
+  private func processImage(item: NSItemProviderReading?) {
+    guard let image = item as? UIImage else {
+      log.error("Failed to read image data")
+      return
+    }
+
+    guard let data = imageToData(image) else {
+      log.error("Failed to process image data")
+      return
+    }
+
+    guard let fileSystem = appContext?.fileSystem else {
+      log.error("Failed to access FileSystem")
+      return
+    }
+
+    let url = URL(fileURLWithPath: fileSystem.cachesDirectory)
+      .appendingPathComponent("clipboard-\(id).\(imageOptions.imageFormat.rawValue)")
+
+    do {
+      try data.write(to: url)
```

Comment:
I thought we ended up with the base64 data for now, to be consistent with the getter. In the future we'll deprecate this and return a reference to the native image instead.